solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -909,15 +907,9 @@ public CoreContainer getCoreContainer() {
   /**
    * Creates a new core and register it in the list of cores. If a core with the
    * same name already exists, it will be stopped and replaced by this one.
-   *
-   * @param dataDir the index directory
-   * @param config  a solr config instance
-   * @param schema  a solr schema instance
-   * @since solr 1.3
    */
-  public SolrCore(CoreContainer coreContainer, String name, String dataDir, SolrConfig config,
-                  IndexSchema schema, NamedList configSetProperties,
-                  CoreDescriptor coreDescriptor, UpdateHandler updateHandler,
+  private SolrCore(CoreContainer coreContainer, String name, ConfigSet configSet, CoreDescriptor coreDescriptor,
```

Comment:
For this one and the other constructors that we changed, can we improve the javadoc to note to the caller that configSet will be used to create the resource loader? That might be too technical for a general audience, but it also seems like something that might cause a debugging headache down the line.

Reply:
I dunno... it's just one detail of a great many details that might be said.  And when you document something you run the risk of reality changing from what you documented and so need to keep in sync.  _Shrug._